partiql-parser/src/main/kotlin/org/partiql/parser/internal/PartiQLParserDefault.kt

```diff
@@ -1546,13 +1574,27 @@ internal class PartiQLParserDefault : PartiQLParser {
             GeneratedParser.MINUS -> Expr.Binary.Op.MINUS
             GeneratedParser.PERCENT -> Expr.Binary.Op.MODULO
             GeneratedParser.CONCAT -> Expr.Binary.Op.CONCAT
-            GeneratedParser.ANGLE_LEFT -> Expr.Binary.Op.LT
+            GeneratedParser.ANGLE_LEFT -> {
+                if (ctx.stop.type == GeneratedParser.ANGLE_RIGHT) Expr.Binary.Op.NE
+                else Expr.Binary.Op.LT
+            }
             GeneratedParser.LT_EQ -> Expr.Binary.Op.LTE
             GeneratedParser.ANGLE_RIGHT -> Expr.Binary.Op.GT
             GeneratedParser.GT_EQ -> Expr.Binary.Op.GTE
-            GeneratedParser.NEQ -> Expr.Binary.Op.NE
+            GeneratedParser.BANG -> Expr.Binary.Op.NE
             GeneratedParser.EQ -> Expr.Binary.Op.EQ
-            else -> throw error(token, "Invalid binary operator")
+            else -> throw error(ctx.start, "Invalid binary operator")
+        }
+
+        private fun convertBinaryMathOp(token: Token) = when (token.type) {
```

Comment:
Taking another look at this, do we need to separate the binary ops between two separate functions? The [PartiQLPigVisitor](https://github.com/partiql/partiql-lang-kotlin/blob/v1-struct-list/partiql-lang/src/main/kotlin/org/partiql/lang/syntax/impl/PartiQLPigVisitor.kt#L1736) just has one binary op conversion function?

Reply:
PIG AST needs to do the get all tokens to have the right error location, hence the list and the fold. 

Error location functionality is not finalized and tested in the new AST. Hence the two separate function to reduce the diff.

Reply:
Thanks for the clarification